### PR TITLE
petsc4py: add petsc4py-factorized solve function

### PR DESCRIPTION
**Pull request by RussTorres**
petsc4py factorized solve implementation is passing tests.  This is based on the code that worked to do single-node rough alignment of 25-point tps, but should also let us do some large multi-node alignments using just the python code and mpiexec (I have not tried this).  I will have this as a draft until I get the setup working, write a comparison test, and document.

### CODE REVIEW

**Comment by djkapner** [comment]
> (I have not tried this). I will have this as a draft until I get the setup working, write a comparison test, and document.

still true?

File: .travis.yml [hunk 1]
```diff
@@ -24,10 +24,26 @@ addons:
...
+- hash -r
+- conda config --set always_yes yes --set changeps1 no
+- conda update -q conda
+- conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION
+- conda activate test-environment
+
 - pip install codecov
 - pip install -r requirements.txt
 - pip install -r test_requirements.txt
+- conda install -c conda-forge petsc4py
```
Comment (djkapner) [inline .travis.yml:46]: just curious, do the pip install instructions not work, or this just brings mpicc along?

File: bigfeta/schemas.py [hunk 2]
```diff
@@ -337,6 +337,11 @@ class BigFetaSchema(ArgSchema):
             description=("tilespec.tforms[i].tform() for i in transform_apply "
                          "will be performed on the matches before matrix "
                          "assembly."))
+    solve_implementation = String(
+        required=False, default="default", missing="default",
+        validator=mm.validate.OneOf(["petsc", "python", "default"]),
```
Comment (djkapner) [inline bigfeta/schemas.py:342]: since they are both python front-ends now, would this make more sense to call then `petsc` and `scipy`?

**Review by djkapner — APPROVED** [review]
some minor comments, but LGTM!
that's great you got this working. I am curious if you've gotten this to work across nodes with mpi

File: .travis.yml (hunk 1 above)
Comment (RussTorres) [reply .travis.yml:46]: I haven't tried too hard, but had some trouble doing `pip install petsc petsc4py` as it says on pypi. The conda recipe seems really nice, though, and the repo technically works without it in `requirements.txt`.

File: bigfeta/schemas.py (hunk 2 above)
Comment (RussTorres) [reply bigfeta/schemas.py:342]: fair point -- I will change this.

**Comment by RussTorres** [comment]
Looked at this a little bit and I think a truly mpi-friendly implementation will probably have to come in another PR.  As I said, I have not had a use case to go beyond a single node yet, but even using this petsc implementation without mpi we are able to do larger solves.

**Pull request merged**